momento-functions/examples/turbopuffer-index-articles.rs

```diff
@@ -0,0 +1,283 @@
+        // to reconstruct the embeddings for our documents
+        for (document, embedding) in chunk.into_iter().zip(embedding_data) {
+            turbopuffer_inputs.push(document.into_turbopuffer_document(embedding.embedding));
+        }
+        // Set to debug if you need to see what is being sent
+        log::trace!("sending to turbopuffer: {}", json!(turbopuffer_inputs));
+        // Send off our transformed data to Turbopuffer, complete with embeddings
+        let result = momento_functions_host::http::post(
+            turbopuffer_endpoint.clone(),
+            [
+                ("Authorization".to_string(), turbopuffer_api_key.clone()),
+                ("Content-Type".to_string(), "application/json".to_string()),
+                ("Accept".to_string(), "*/*".to_string()),
+                (
+                    "User-Agent".to_string(),
+                    "momento-turbobuffer-example".to_string(),
+                ),
+            ],
+            json!({
+                "upsert_rows": turbopuffer_inputs,
+                "distance_metric": "cosine_distance",
+            }),
+        );
+        match result {
+            Ok(response) => {
+                log::debug!(
+                    "turbopuffer response: {} - {:?}",
+                    response.status,
+                    response.headers
+                );
+                if response.status != 200 {
+                    let message = format!(
+                        "Failed to index documents: {}",
+                        String::from_utf8(response.body).unwrap_or_default(),
+                    );
+                    return Ok(WebResponse::new().with_status(response.status).with_body(
+                        json!({
+                            "message": message,
+                        }),
+                    )?);
+                }
+            }
+            Err(e) => {
+                log::error!("Failed to index documents: {e:?}");
+                return Ok(WebResponse::new().with_status(500).with_body(json!({
+                    "message": e.to_string(),
+                }))?);
+            }
+        }
+    }
+    Ok(WebResponse::new().with_status(200).with_body(json!({
+        "message": "Documents indexed successfully",
+    }))?)
+}
+
+// ------------------------------------------------------
+// | Utility functions for convenience
+// ------------------------------------------------------
+
+fn get_embeddings(mut documents: Vec<String>, openai_key: String) -> WebResult<Vec<EmbeddingData>> {
+    log::debug!("getting embeddings for input");
+    for document in &mut documents {
+        if document.contains("\n") {
+            // openai guide currently says to replace newlines with spaces. This, then, must be how you get the cargo to come.
+            // https://platform.openai.com/docs/guides/embeddings
+            *document = document.replace("\n", " ");
+        }
```

Comment:
I would just remove this part.

Reply:
The comment? We do have to replace the newlines for the embeddings

Reply:
Ah gotcha, fixed in https://github.com/momentohq/functions/pull/4/commits/d1e243e41076946b3443c66b701cdcfb88d63e2d